### DISCUSSION TITLE
AttributeError: 'AsyncSession' object has no attribute 'fetch_one'

### DISCUSSION THREAD

**Discussion by iBrokeYourGui** [fastapi-users/fastapi-users#900] [Q&A] [answered]
Hi! 
I just copy exemple for sqlAlchemy and have some trobles with database: 
1) Migration:
I dont know when i shoul use async def create_db_and_tables(): ? 
I create migration for alembic like that sqlite:///./test.db, and it work fine, but before 'upgrade head' i need import fastapi_users_db_sqlalchemy to migration shcema, couse it not importing autamaticly. 

2) When i try register or another get request a have error from 
async def get_user_db(session: AsyncSession = Depends(get_async_session)):
    yield SQLAlchemyUserDatabase(UserDB, session, UserTable)
FIrst i get: AttributeError: type object 'UserTable' has no attribute 'select'

if i writeUserTable as UserTable.__table__, i'll get next error
AttributeError: 'AsyncSession' object has no attribute 'fetch_one'
Here's i don't know what is problem is (((

**Comment by frankie567** [comment, marked as answer]
I've made a number of fixes to the SQLAlchemy examples. The problems you have are caused by two things:

* If you setup Alembic, you don't need `create_db_and_tables`. It's a shortcut to create tables quickly while development, but it's not needed with a proper migration system.

https://github.com/fastapi-users/fastapi-users/blob/9b1f9bbd0fced9d24c87c4175e280f0a8abfead5/examples/sqlalchemy/app/app.py#L31-L34

* The error you have is because the example is designed to work with the newest version of our SQLAlchemy adapter, but the old one was installed. To fix this, simply install `fastapi-users[sqlalchemy2]` in your dependencies (**not** `fastapi-users[sqlalchemy]`)

https://github.com/fastapi-users/fastapi-users/blob/9b1f9bbd0fced9d24c87c4175e280f0a8abfead5/examples/sqlalchemy/requirements.txt#L1-L3

**Reply by iBrokeYourGui** [reply]
Great thanks! 
DB work correcn now!
